tests/src/utils/test_rlp.cairo

```diff
@@ -16,16 +16,96 @@ func test__decode{range_check_ptr}(output_ptr: felt*) {
     %}
 
     // When
-    let (local items: RLP.Item*) = alloc();
-    RLP.decode(data_len, data, items);
+    let (items_len, items) = RLP.decode(data_len, data);
+
+    %{
+        def flatten_list(address, len, output_offset, list_level):
```

Comment:
the signature is clearer below ihmo

```
flatten_list(data_ptr, data_len, output_offset)
```

you don't need the `list_level` neither

Reply:
the recursion would be easier and more cairo like by giving not the `output_offset` but the `output_ptr` offset